src/Neo/Plugins/Plugin.cs

```diff
@@ -247,33 +247,46 @@ protected internal virtual void OnSystemLoaded(NeoSystem system)
         /// <returns><see langword="true"/> if the <paramref name="message"/> is handled by a plugin; otherwise, <see langword="false"/>.</returns>
         public static bool SendMessage(object message)
         {
+            foreach (var plugin in Plugins)
+            {
+                if (plugin.IsStopped)
+                {
+                    continue;
+                }
 
-            return Plugins.Any(plugin =>
+                bool result;
+                try
                 {
-                    try
-                    {
-                        return !plugin.IsStopped &&
-                               plugin.OnMessage(message);
-                    }
-                    catch (Exception ex)
+                    result = plugin.OnMessage(message);
+                }
+                catch (Exception ex)
+                {
+                    Utility.Log(nameof(Plugin), LogLevel.Error, ex);
+
+                    switch (plugin.ExceptionPolicy)
                     {
-                        switch (plugin.ExceptionPolicy)
-                        {
-                            case UnhandledExceptionPolicy.StopNode:
-                                throw;
-                            case UnhandledExceptionPolicy.StopPlugin:
-                                plugin.IsStopped = true;
-                                break;
-                            case UnhandledExceptionPolicy.Ignore:
-                                break;
-                            default:
-                                throw new InvalidCastException($"The exception policy {plugin.ExceptionPolicy} is not valid.");
-                        }
-                        Utility.Log(nameof(Plugin), LogLevel.Error, ex);
-                        return false;
+                        case UnhandledExceptionPolicy.StopNode:
+                            throw;
+                        case UnhandledExceptionPolicy.StopPlugin:
+                            plugin.IsStopped = true;
+                            break;
+                        case UnhandledExceptionPolicy.Ignore:
+                            break;
+                        default:
+                            throw new InvalidCastException($"The exception policy {plugin.ExceptionPolicy} is not valid.");
```

Comment:
```suggestion
                            throw new InvalidCastException($"The exception policy {plugin.ExceptionPolicy} is not valid.", ex);
```

Put real exception as an inter exception.